<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Main_SAM_test</class>
 <widget class="QMainWindow" name="Main_SAM_test">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>655</width>
    <height>417</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Main_SAM_test</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>651</width>
      <height>351</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>SAMs</string>
     </attribute>
     <widget class="QPushButton" name="buttonTest">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>50</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Test</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>ID</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMTorque">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>50</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <property name="maximum">
       <number>4</number>
      </property>
     </widget>
     <widget class="QDial" name="dialPos">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>130</y>
        <width>171</width>
        <height>151</height>
       </rect>
      </property>
      <property name="minimum">
       <number>401</number>
      </property>
      <property name="maximum">
       <number>3700</number>
      </property>
      <property name="value">
       <number>1800</number>
      </property>
     </widget>
     <widget class="QPushButton" name="Buttonconnect">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>20</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Connect</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Torque</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMID">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <property name="maximum">
       <number>25</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>220</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>KD</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>140</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Read PID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>140</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Write PID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>160</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>KP</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>190</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>KI</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMGetPID">
      <property name="geometry">
       <rect>
        <x>499</x>
        <y>250</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get PID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMP">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>160</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMD">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>220</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMI">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>190</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMSetPID">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>250</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set PID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMGetPos">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>59</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Pos12</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMSetPos">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>71</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Pos12</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMGetPos">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>71</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Pos12</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMSetPos">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>110</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="minimum">
       <number>401</number>
      </property>
      <property name="maximum">
       <number>3700</number>
      </property>
      <property name="value">
       <number>1500</number>
      </property>
     </widget>
     <widget class="QDial" name="dialPos8bit">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>130</y>
        <width>171</width>
        <height>151</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>254</number>
      </property>
      <property name="value">
       <number>254</number>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>270</y>
        <width>59</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>12 bits</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>270</y>
        <width>59</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>8 bits</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMSetPos8">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>110</y>
        <width>71</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Pos8</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMGetPos8">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>80</y>
        <width>71</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Pos8</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMSetPos8">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>110</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>254</number>
      </property>
      <property name="value">
       <number>254</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMGetPos8">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>80</y>
        <width>59</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Pos8</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMSetPassive">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>280</y>
        <width>81</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Passive</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_SAMGetTorq8">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>80</y>
        <width>59</width>
        <height>14</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Torq8</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMP">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>160</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMI">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>190</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_SAMD">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>220</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMAllGetPos8">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>20</y>
        <width>101</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get All Pos8</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_SAMGetAllPos12">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>50</y>
        <width>101</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get All Pos12</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Pose</string>
     </attribute>
     <widget class="QTableWidget" name="tableWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>211</width>
        <height>251</height>
       </rect>
      </property>
      <row>
       <property name="text">
        <string>New Row</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>New Row</string>
       </property>
      </row>
      <column>
       <property name="text">
        <string>New Column</string>
       </property>
      </column>
     </widget>
     <widget class="QPushButton" name="pushButton_3">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>90</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Pose</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_4">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>30</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Test Pose</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_5">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>60</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Get Pose</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="tableWidget_2">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>10</y>
        <width>171</width>
        <height>251</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_6">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>120</y>
        <width>111</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Init Position</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>30</y>
        <width>66</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Motion</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_7">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>40</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Sit down</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_8">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>80</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>stand up</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_9">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>120</y>
        <width>91</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Walk 1 Step</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_10">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>160</y>
        <width>91</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Walk 4 step</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Page</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>655</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
   </widget>
   <addaction name="menu_File"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
